### PR TITLE
[stdlib] Make `object` `Representable` and `Formattable`

### PR DESCRIPTION
**Pull request by bgreni**
Make `object` `Representable` and `Formattable`.

### CODE REVIEW

File: stdlib/src/builtin/object.mojo [hunk 1]
```diff
@@ -557,18 +563,22 @@ struct _ObjectImpl(CollectionElement, CollectionElementNew, Stringable):
         else:
             lhs = lhs.convert_bool_to_int()
 
-    fn __str__(self) -> String:
+    fn format_to(self, inout writer: Formatter):
         """Returns the name (in lowercase) of the specific object type."""
```
Comment (gryznar) [inline stdlib/src/builtin/object.mojo:567]: This docstring is no longer valid

File: stdlib/src/builtin/object.mojo [hunk 2]
```diff
@@ -577,32 +587,47 @@ struct _ObjectImpl(CollectionElement, CollectionElementNew, Stringable):
...
                 + str(object(entry[].value.copy()))
             )
             print_sep = True
-        res += "}"
-        return res
+        writer.write("}")
+        return
+
+    fn __repr__(self) -> String:
+        """Returns the name (in lowercase) of the specific object type."""
```
Comment (gryznar) [inline stdlib/src/builtin/object.mojo:622]: (in lowercase)? `None` is represented as `"None"` and not `"none"`. The same applies to `fn __str__`

File: stdlib/src/builtin/object.mojo [hunk 3]
```diff
@@ -577,32 +587,47 @@ struct _ObjectImpl(CollectionElement, CollectionElementNew, Stringable):
...
+        return self.__str__()
+
+    fn __str__(self) -> String:
+        """Returns the name (in lowercase) of the specific object type."""
+        var output = String("")
+        var writer = output._unsafe_to_formatter()
+        self.format_to(writer)
+        return output
```
Comment (ConnorGray) [inline stdlib/src/builtin/object.mojo:630]: **Suggestion** We can avoid the need to touch the internal formatter functions if we change this to:

```mojo
fn __str__(self) -> String:
    return String.format_sequence(self)
```

**Comment by ConnorGray** [comment]
Thanks for this improvement @bgreni! 🙂 

I left a couple of minor pieces of feedback, and I also agree with @gryznar's suggestions about a couple of doc strings.

Once those are taken care of, I'll proceed with syncing this in 🙂

File: stdlib/src/builtin/object.mojo [hunk 4]
```diff
@@ -943,6 +979,16 @@ struct object(IntableRaising, ImplicitlyBoolable, Stringable):
         """
         return str(self._value)
```
Comment (soraros) [inline stdlib/src/builtin/object.mojo:980]: I think we also need to move this `String.format_sequence`?
```mojo
fn __str__(self) -> String:
  return String.format_sequence(self)
```

**Comment by ConnorGray** [comment]
!sync

**Pull request closed without merging**